### DISCUSSION TITLE
rke2-server service start failing due to failed to get unit file state nm-cloud-setup.service: No such file or directory

### DISCUSSION THREAD

**Discussion by anishagoyal0504** [rancher/rke2#6418] [General]
I have downloaded the cilium images, rke2-core-linux-amd64 and rke2.linux-amd64 on  my controlplane node and installed rke2 binary (rke2.linux-amd64), at somehow I have followed the rke2 instructions for Ari-Gap environment however rke2-server service is getting failed due to the below: 

Process: 1234566 ExecStart=/usr/local/bin/rke2 server (code=exited, status=1/FAILURE)
Process: 1234566 ExecStartPre=/bin/sh -xc ! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service (code=exited, status=0/SUCCESS)
rke2-server service start failing due to failed to get unit file state nm-cloud-setup.service: No such file or directory

I have checked the service nm-cloud-setup.service does not exist at all, not sure why it is still pointing to this service. I tried to google it but could not find the correct answer.

Any suggestion would be great

**Comment by brandond** [comment]
> rke2-server service start failing due to failed to get unit file state nm-cloud-setup.service: No such file or directory

No, that's not what's happening. You're misreading the logs - the ExecStartPre that checks to ensure the `nm-cloud-setup.service`is not enabled is exiting successfully, because the service does not exist.

Check the rest of the logs to see why the `rke2 server` command is exiting.

**Reply by anishagoyal0504** [reply]
i restarted the rke2 server and trying to start the rke2-servver.service, it just stuck on the black screen, I have to exit by CTR+C to get the prompt back. Upon looking at systemctl rke2-server logs, I am getting this messages or warnings:

Waiting to retrieve kube-proxy configuration: Serevr is not ready : https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal server Error
warning msg: Failed to list nodes with etcd role: runtime core not ready

to implement rke2 in Airgap environment seems to be causing many issues on redhat 8 linux server, with shell script seems to be easy with internet access

**Reply by brandond** [reply]
You'll need to look further at the logs. I suspect you're missing some images that RKE2 needs to start up. Have you checked the docs on installing airgap image tarballs?

**Reply by anishagoyal0504** [reply]
I followed through rke2 doc for air gap, not a well written document so I downloaded rke2 release from their release page. All the images are located in tar.gz compressed format on /var/lib/rancher/rke2/agent/images/rke2-images-core.linux-amd64.tar.gz

Have you tried yourself installing tar files on redhat 8 linux?
Any other view you think,

**Reply by brandond** [reply]
It is normal to see some "not ready" messages while everything starts up. If this lasts longer than a few minutes, check the containerd.log and kubelet.log to see why the static pods are not starting. You might also check /var/log/pods for errors.

**Reply by anishagoyal0504** [reply]
By looking at kubelet log, I found some messages "kubelet_node_status: "unable to register node with API server" err=POST \"https://127.0.0.1:6443/api/v1/nodes": connect: connection refused" node=<"rke2-server-controlplane-nodename>
Eviction_manager: failed to get summary stats" err="failed to get node info: node\<"rke2-server-controlplane-nodename"> not found

what to do now? please guide me

**Reply by brandond** [reply]
There are going to be a lot of errors until the apiserver comes up. You are welcome to zip up the logs and upload them here, but I'm not able to hold your hand through troubleshooting an unspecified problem.

**Reply by anishagoyal0504** [reply]
I have fixed the rke2-server.service issue, now it is running however, when I did "export KUBECONFIG=/etc/rancher/rke2/rke2.yaml PATH=$PATH:/var/lib/rancher/rke2/bin" as per rke2 doc however when I run kubectl get nodes, it says "kubectl not found" I can see the kubectl is located under /usr/local/bin/kubectl 

In addition to this, I have kubeconfig file configured as below:
mkdir -p $HOME/.kube
sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
sudo chown $(id -u):$(id -g) $HOME/.kube/config

**Reply by brandond** [reply]
RKE2 does not provide `/etc/kubernetes/admin.conf`. Did you want to use the one at `/etc/rancher/rke2/rke2.yaml`?

> I can see the kubectl is located under /usr/local/bin/kubectl

ok, so put that in your path?

**Reply by anishagoyal0504** [reply]
Finally, it is working, It was mix of permission issue on the bin/kubectl and rke2.yaml, thanks for your help @brandond

**Reply by anishagoyal0504** [reply]
> RKE2 does not provide `/etc/kubernetes/admin.conf`. Did you want to use the one at `/etc/rancher/rke2/rke2.yaml`?
> 
> > I can see the kubectl is located under /usr/local/bin/kubectl
> 
> ok, so put that in your path?

May be you missed the last comment, it is all working.

to add another nodes as rke2-server to join the cluster, do I have to instrall the followings:

- rke2-images-core.linux-amd64.tar.gz
- rke2-images-cilium.linux-amd64.tar.gz
- rke2-.linux-amd64.tar.gz (rke2 binary)

or  just install rke2 binary which is rke2-.linux-amd64.tar.gz on other controlplane node so that they can be treated as rke2-server.

Please advice

**Comment by anishagoyal0504** [comment]
to add another nodes as rke2-server to join the cluster, do I have to instrall the followings:

- rke2-images-core.linux-amd64.tar.gz
- rke2-images-cilium.linux-amd64.tar.gz
- rke2-.linux-amd64.tar.gz (rke2 binary)

or  just install rke2 binary which is rke2-.linux-amd64.tar.gz on other controlplane node so that they can be treated as rke2-server.

Please advice

**Reply by brandond** [reply]
If you don't have an external registry mirror available, all the nodes need images to be available locally via tarballs. You should put the same set of images on all nodes.

Alternately, you can enable the embedded registry mirror to allow sharing images between nodes. This is not currently covered in the RKE2 docs, but you can reference the k3s docs at https://docs.k3s.io/installation/registry-mirror

**Reply by anishagoyal0504** [reply]
Do I have to install only rke2-images-core.linux-amd64.tar.gz, rke2-.linux-amd64.tar.gz (rke2 binary) on other controlplane nodes in AirGap environment.

Same thing with RKE2-Agent, I have to install the images and rke2 binary. Thanks

**Reply by brandond** [reply]
As I said, all the nodes need all the images. Why wouldn't you need the CNI on agents?
